features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Overview.jsx

```diff
@@ -136,26 +141,66 @@ class Overview extends Component {
     handleTagChange(tags) {
         const api = new Api();
         const { apiUUID } = this;
+        const currentAPI = this.state.api;
         const promisedApi = api.get(apiUUID);
         promisedApi
-            .then((response) => {
-                const apiData = JSON.parse(response.data);
+            .then((getResponse) => {
+                const apiData = JSON.parse(getResponse.data);
                 apiData.tags = tags;
                 const promisedUpdate = api.update(apiData);
-                promisedUpdate.catch((errorResponse) => {
-                    console.error(errorResponse);
-                    Alert.error('Error occurred while updating tags');
-                });
+                promisedUpdate
+                    .then((updateResponse) => {
+                        this.setState({ api: updateResponse.obj });
+                    })
+                    .catch((errorResponse) => {
+                        console.error(errorResponse);
+                        Alert.error('Error occurred while updating tags');
+                        this.setState({ api: getResponse.obj });
```

Comment:
`getResponse.obj` will lead to undefined if the exception is not a swaggerjs request failure.

Reply:
which one should be used here?

Reply:
As pert the [doc](https://github.com/swagger-api/swagger-js) it's better to use the `.body` property, .obj is for legacy purposes